lib/src/util.dart

```diff
@@ -15,4 +18,24 @@ class Util {
         ? truncatedDivision.toString()
         : '$truncatedDivision.$paddingZeros$remainder';
   }
+
+  /// Convert a proto [Date] instance to Dart DateTime.
+  static DateTime convertProtoDateToDateTime(
+          InsertionOrder_Budget_BudgetSegment_DateRange_Date date) =>
+      DateTime(date.year, date.month, date.day);
+
+  /// Convert a string date(yyyy/mm/dd) to Dart DateTime.
+  static DateTime convertStringDateToDateTime(String date) {
+    return DateFormat('yyyy/MM/dd').parse(date);
+  }
+
+  /// Return `true` if [target] is before [start] and after [end],
+  /// false otherwise.
+  static bool isBetweenDates(DateTime target, DateTime start, DateTime end) =>
+      (target.isAfter(start) || target.isAtSameMomentAs(start)) &&
```

Comment:
good consideration of edge cases.
this may be a good place to drop a bug indication that we are not handling timezones. Are the start/stop timezones at the same timezone as reporting api?

Reply:
Reporting api takes ms since unix epoch as input, so I think the target/start/end here are all at the same timezones since I converted the flight start/end to ms and then pass them to the reporting api. 

but isBetweenDates(DateTime.now(), flight start, flight end) will have issues, since DateTime.now() returns current date and time in the local time zone, but public api returns flight start/end in advertiser time zone.